specification/storagecache/resource-manager/Microsoft.StorageCache/preview/2020-08-13-aml-preview/storagecache.json

```diff
@@ -0,0 +1,2618 @@
+        "nextLink": {
+          "description": "The URI to fetch the next page of Storage Targets.",
+          "type": "string"
+        },
+        "value": {
+          "description": "The list of Storage Targets defined for the Cache.",
+          "items": {
+            "$ref": "#/definitions/StorageTarget"
+          },
+          "type": "array"
+        }
+      },
+      "type": "object"
+    },
+    "URLString": {
+      "description": "A fully qualified URL.",
+      "type": "string"
+    },
+    "UsageModel": {
+      "description": "A usage model.",
+      "properties": {
+        "display": {
+          "description": "Localized information describing this usage model.",
+          "type": "object",
+          "properties": {
+            "description": {
+              "description": "String to display for this usage model.",
+              "type": "string"
+            }
+          }
+        },
+        "modelName": {
+          "description": "Non-localized keyword name for this usage model.",
+          "type": "string"
+        },
+        "targetType": {
+          "description": "The type of Storage Target to which this model is applicable (only nfs3 as of this version).",
+          "type": "string"
+        }
+      },
+      "type": "object"
+    },
+    "UsageModelsResult": {
+      "description": "A list of Cache usage models.",
+      "properties": {
+        "nextLink": {
+          "description": "The URI to fetch the next page of Cache usage models.",
+          "type": "string"
+        },
+        "value": {
+          "description": "The list of usage models available for the subscription.",
+          "items": {
+            "$ref": "#/definitions/UsageModel"
+          },
+          "type": "array"
+        }
+      },
+      "type": "object"
+    },
+    "amlFilesystem": {
+      "description": "An amlFilesystem instance. Follows Azure Resource Manager standards: https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md",
+      "properties": {
+        "tags": {
+          "description": "ARM tags as name/value pairs.",
+          "type": "object"
+        },
+        "id": {
+          "description": "Resource ID of the amlFilesystem.",
+          "readOnly": true,
+          "$ref": "#/definitions/URLString"
+        },
+        "location": {
+          "description": "Region name string.",
+          "type": "string"
+        },
+        "name": {
+          "description": "Name of amlFilesystem.",
+          "readOnly": true,
+          "$ref": "#/definitions/amlFilesystemName"
+        },
+        "type": {
+          "description": "Type of the filesystem; Microsoft.StorageCache/amlFilesystem",
+          "readOnly": true,
+          "type": "string"
+        },
+        "identity": {
+          "$ref": "#/definitions/IdentityProperties",
+          "description": "The managed identity used by the amlFilesystem, if configured."
+        },
+        "properties": {
```

Comment:
Every amlFilesystem is uniquely identified by a UUID. It'd be very helpful to expose that here for testing and support. Doing so is harmless. CRP does this for VMs, so there's precedent.

Reply:
Rather than exposing internal information to the customers, you should consider a call similar to our GetCachePlus which is specifically for Support and Dev use and exposes all sorts of useful info (it's a Jarvis action).   One advantage of taking that path is that you can expose sensitive internal details, as the jarvis actions are securely protected.

Reply:
I think that's true in general, but the UUID is not sensitive, and other RPs expose this.

Reply:
Added.

Reply:
The validator was not happy with the uuid.

{
  "type": "Warning",
  "code": "GuidUsage",
  "message": "Guid used in model definition 'amlFilesystem' for property 'uuid'. Usage of Guid is not recommanded. If GUIDs are absolutely required in your service, please get sign off from the Azure API review board.",
  "id": "R3017",
  "validationCategory": "ARMViolation",
  "providerNamespace": null,
  "resourceType": null,
  "sources": [
    "file:///C:/lustre/azure-rest-api-specs/specification/storagecache/resource-manager/Microsoft.StorageCache/preview/2020-08-13-aml-preview/storagecache.json:2244:6 ($.definitions.amlFilesystem.properties)"
  ],
  "jsonref": "file:///C:/lustre/azure-rest-api-specs/specification/storagecache/resource-manager/Microsoft.StorageCache/preview/2020-08-13-aml-preview/storagecache.json:2244:6 ($.definitions.amlFilesystem.properties)",
  "json-path": "file:///C:/lustre/azure-rest-api-specs/specification/storagecache/resource-manager/Microsoft.StorageCache/preview/2020-08-13-aml-preview/storagecache.json:2244:6 ($.definitions.amlFilesystem.properties)"
}

Reply:
Took it back out of the latest commit.

Reply:
Fascinating. OK... not gonna try to swim up that stream.